### PR TITLE
[py] convenience api to get cerr/cout vw output

### PR DESCRIPTION
**Pull request by lalo**
there was a convo in gitter that prompted this work.

### CODE REVIEW

File: python/vowpalwabbit/pyvw.py [hunk 1]
```diff
@@ -500,6 +511,17 @@ def finish(self):
             pylibvw.vw.finish(self)
             self.finished = True
 
+    def get_log(self):
```
Comment (lalo) [inline python/vowpalwabbit/pyvw.py:514]: get_summary() might be a better name

**Comment by lalo** [comment]
depends on #2750

**Comment by jackgerrits** [comment]
@lalo is this still valid after the reverted PR?

**Comment by lalo** [comment]
yes it is

**Review by jackgerrits — APPROVED** [review]

File: python/pylibvw.cc [hunk 2]
```diff
@@ -57,10 +57,46 @@ const size_t pPDF = 10;
 
 void dont_delete_me(void* arg) {}
 
-vw_ptr my_initialize(std::string args)
+class python_log_wrapper
```
Comment (lalo) [inline python/pylibvw.cc:60]: expose this one

File: python/pylibvw.cc [hunk 3]
```diff
@@ -57,10 +57,46 @@ const size_t pPDF = 10;
...
 {
   if (args.find_first_of("--no_stdin") == std::string::npos) args += " --no_stdin";
-  vw* foo = VW::initialize(args);
+
+  trace_message_t trace_listener = nullptr;
+  void* trace_context = nullptr;
+
+  if (!py_log.is_none())
+  {
+    // leaking memory for now
```
Comment (lalo) [inline python/pylibvw.cc:92]: pending fix leak

**Pull request merged**